gobblin-core/src/main/java/gobblin/writer/http/AsyncHttpWriterBuilder.java

```diff
@@ -44,11 +52,12 @@
  * @param <RQ> type of request
  * @param <RP> type of response
  */
+@Slf4j
 public abstract class AsyncHttpWriterBuilder<D, RQ, RP> extends FluentDataWriterBuilder<Void, D, AsyncHttpWriterBuilder<D, RQ, RP>> {
   public static final String CONF_PREFIX = "gobblin.writer.http.";
 
   @Getter
-  protected State state;
+  protected WorkUnitState state;
```

Comment:
Let's not weaken this abstraction, keeping it as `State`. What we want here is a way to create a broker. Could we introduce a protected function `SharedResourcesBroker createBroker()`? The default implementation is to cast `State` to `WorkUnitState` and get its broker.